# csi_rover_controller_joint_topics.yaml
csi_rover:


  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 30

  # steering
  left_front_steering_controller:
    joint: left_front_steering_joint
    type: effort_controllers/JointPositionController
    pid: {p: 100.0, i: 0.01, d: 10.0} #pid: {p: 4.0, i: 0.0, d: 1.0}
  right_front_steering_controller:
    joint: right_front_steering_joint
    type: effort_controllers/JointPositionController
    pid: {p: 100.0, i: 0.01, d: 10.0} #pid: {p: 4.0, i: 0.0, d: 1.0}
  left_rear_steering_controller:
    joint: left_rear_steering_joint
    type: effort_controllers/JointPositionController
    pid: {p: 100.0, i: 0.01, d: 10.0} #pid: {p: 4.0, i: 0.0, d: 1.0}
  right_rear_steering_controller:
    joint: right_rear_steering_joint
    type: effort_controllers/JointPositionController
    pid: {p: 100.0, i: 0.01, d: 10.0} #pid: {p: 4.0, i: 0.0, d: 1.0}

  # wheel rotation
  left_front_axle_controller:
    type: effort_controllers/JointVelocityController
    joint: left_front_axle
    pid: {p: 1.5, i: 1.0, d: 0.0, i_clamp: 10.0}
  right_front_axle_controller:
    type: effort_controllers/JointVelocityController
    joint: right_front_axle
    pid: {p: 1.5, i: 1.0, d: 0.0, i_clamp: 10.0}
  left_rear_axle_controller:
    type: effort_controllers/JointVelocityController
    joint: left_rear_axle
    pid: {p: 1.5, i: 1.0, d: 0.0, i_clamp: 10.0}
  right_rear_axle_controller:
    type: effort_controllers/JointVelocityController
    joint: right_rear_axle
    pid: {p: 1.5, i: 1.0, d: 0.0, i_clamp: 10.0}





















